src/basic/fs-util.c

```diff
@@ -1028,3 +1029,44 @@ int parse_cifs_service(
 
         return 0;
 }
+
+char *resolve_auto_dir(const char *path) {
+        _cleanup_free_ char *parent = NULL;
+        _cleanup_free_ char *glob_path = NULL;
+        _cleanup_globfree_ glob_t pglob = {};
+        int r;
+
+        if (!path)
+                return NULL;
+
+        log_debug("Trying to version %s.", path);
+        r = path_extract_directory(path, &parent);
+        if (r < 0 || !endswith(parent, "auto.d"))
+                goto error;
+
+        r = asprintf(&glob_path, "%s*", path);
```

Comment:
this will fall appart if people do this:

```
systemd-nspawn --directory='/srv/my***images.auto.d' -b
```

i.e. people have a path with an asterisk, and they explicitly quoted it so that the asterisks aren't understood as globs. Yet here they will be interpreted nonetheless. Hence: please do not use globs for internal enumerations.

Reply:
Ouch. That should apply to all cases where the path comes from untrusted source and there's some kind of trust boundary? Luckily there doesn't seem to be any CVEs related to [glob](https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=glob), but could you be the first one to notice this?

Reply:
Not using globs anymore